### PR TITLE
Edit designate install guide

### PR DESCRIPTION
**Pull request by svetkars**
Edited the designate guide.

### CODE REVIEW

File: doc/installing/index.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+.. _desginate-ug:
```
Comment (robb-romans) [inline doc/installing/index.rst]:
```suggestion
.. _designate-ug:
```

File: doc/installing/index.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+by OpenStack (RPCO) and Rackspace
+Private Cloud powered by RedHat@ (RPCR) environments.
+This document is intended for Rackspace deployment and support engineers
+who need to install, configure, and operate OpenStack designate in
+RPC and RPCR environments.
+
+.. toctree::
+   :maxdepth: 2
+
+   common/common-front.rst
```
Comment (robb-romans) [inline doc/installing/index.rst]:
```suggestion
   ../common/common-front.rst
```

File: doc/installing/index.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+   prepare-installation.rst
+   configure-dns-server-pool.rst
+   deploy-designate.rst
+   common/common-end.rst
```
Comment (robb-romans) [inline doc/installing/index.rst]:
```suggestion
   ../common/common-end.rst
```

File: doc/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Contributor guidelines
+
+These guidelines provide the general process for maintaining source code for the instructions and templates in the docs-starter-kit repository (repo).
```
Comment (cyrichardson) [inline doc/CONTRIBUTING.md]: instructions and templates in this repository  (repo).

File: doc/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+- [Repo description](#repo-description)
+- [Updating and adding content](#updating-and-adding-content)
+- [Using writing guidelines](#using-writing-guidelines)
+- [Submitting your content](#submitting-changes)
+- [Previewing changes](#previewing-changes)
+
+## Repo description
```
Comment (cyrichardson) [inline doc/CONTRIBUTING.md]: Maybe delete this section??

File: doc/GITHUBING.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+[Generate SSH keys](https://help.github.com/articles/generating-ssh-keys/)
+
+## Workflow
+
+1. Create a fork of this repo.
+
+2. Use the SSH URL for the forked repo to clone it to your local system:
+
+    ```
+    git clone git@github.rackspace.com:my-github-username/docs-starter-kit.git
```
Comment (cyrichardson) [inline doc/GITHUBING.md]: change to your repo

File: doc/GITHUBING.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    ```
+
+3. Track the upstream repo:
+
+    ```
+    cd docs-starter-kit
```
Comment (cyrichardson) [inline doc/GITHUBING.md]: Change this line and the next to be for your repo.

File: doc/_deconst.json [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{
+  "contentIDBase": "https://github.com/rackerlabs/docs-repo-template/user-guide/",
```
Comment (cyrichardson) [inline doc/_deconst.json]: Update this line and the next one for your repo.

File: doc/conf.py [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+       formatted = EventHandlers.CURRENT_VERSION
+except TypeError:
+    formatted = 'v1.0.0'
+
+version = '{}'.format(formatted)
+
+# Global variables that are replaced by the specified value during the build
+# process.
+
+rst_epilog = """
```
Comment (cyrichardson) [inline doc/conf.py]: Not sure that you use these variables, but if you do you would need to modify them.

File: doc/conf.py [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+    'rax-cloud': ('http://www.rackspace.com/cloud/%s', ''),
+    'rax-dev': ('https://developer.rackspace.com/%s', ''),
+    'rax-devdocs': ('http://developer.rackspace.com/docs/%s', ''),
+    'rax-devguide': ('http:/developer.rackspace.com/docs/%s', ''),
+    'rax-api':
+    ('http:/developer.rackspace.com/docs/%s/api-reference', ''),
+    'rax-git': ('https://github.com/rackspace/%s', ''),
+    'mycloud': ('https://mycloud.rackspace.com/%s', ''),
+    'rax-glossary': ('https://developer.rackspace.com/docs/glossary/%s', ''),
+    'mycloud': ('https://mycloud.rackspace.com/%s', ''),
```
Comment (cyrichardson) [inline doc/conf.py]: Will need to update to new URL - https://login.rackspace.com

File: doc/conf.py [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+    'rax-cart': ('http://cart.rackspace.com/%s', ''),
+    'rax-special': ('http://%s.rackspace.com/', ''),
+    'rax-cloud': ('http://www.rackspace.com/cloud/%s', ''),
+    'rax-dev': ('https://developer.rackspace.com/%s', ''),
+    'rax-devdocs': ('http://developer.rackspace.com/docs/%s', ''),
+    'rax-devguide': ('http:/developer.rackspace.com/docs/%s', ''),
+    'rax-api':
+    ('http:/developer.rackspace.com/docs/%s/api-reference', ''),
+    'rax-git': ('https://github.com/rackspace/%s', ''),
+    'mycloud': ('https://mycloud.rackspace.com/%s', ''),
```
Comment (cyrichardson) [inline doc/conf.py]: Will need to update to new URL - https://login.rackspace.com

File: doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+Designate Installation Guide
+============================
+
+*Last updated:* |today|
+
+.. Define |product name| in conf.py
+.. You can define other variables in that file and use them throughout
+   your documents.
+
+``rpc-designate`` simplifies installation of multi-tenant
```
Comment (cyrichardson) [inline doc/index.rst]: So this will be RPC Designate?

multitenant

File: doc/index.rst [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+DNS as a Service in the Rackspace Private Cloud powered
+by OpenStack (RPCO) and Rackspace
+Private Cloud powered by RedHat@ (RPCR) environments.
+This document is intended for Rackspace deployment and support engineers
+who need to install, configure, and operate OpenStack designate in
+RPC and RPCR environments.
```
Comment (cyrichardson) [inline doc/index.rst]: RPCO and RPCR

File: doc/installing/configure-dns-server-pool.rst [hunk 14]
```diff
@@ -0,0 +1,78 @@
+.. _configure-dns-server-pool:
+
+Configure a DNS server pool
+===========================
+
+Designate supports multiple pools of DNS servers which enable network
```
Comment (cyrichardson) [inline doc/installing/configure-dns-server-pool.rst]: that enable

File: doc/installing/configure-dns-server-pool.rst [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+     - A list of the name server (NS) records for the
+       zones that are hosted within this pool.
+   * - ``nameservers``
+     - .. code-block:: bash
+
+          nameservers:
+          - host: 127.0.0.1
+            port: 53
+
+     - A list out all name servers to test against. The servers
```
Comment (cyrichardson) [inline doc/installing/configure-dns-server-pool.rst]: list of

File: doc/installing/configure-dns-server-pool.rst [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+          options:
+            host: 127.0.0.1
+            port: 53
+            rndc_host: 127.0.0.1
+            rndc_port: 953
+            rndc_key_file: /etc/designate/rndc.key
+
+     - The ``targets`` section is where you define the target name server on
+       which you perform actions. For each external DNS server, you need add
+       one of the following sections.
```
Comment (cyrichardson) [inline doc/installing/configure-dns-server-pool.rst]: the following information:

File: doc/installing/deploy-designate.rst [hunk 17]
```diff
@@ -0,0 +1,29 @@
+.. _deploy-designate:
+
+Deploy designate
```
Comment (cyrichardson) [inline doc/installing/deploy-designate.rst]: RPCO Designate.

File: doc/installing/deploy-designate.rst [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+================
+
+After you complete the steps described in
+:ref:`prepare-installation` and
+:ref:`configure-dns-server-pool`, you can deploy designate on your
+RPCO or RPCR environment.
+
+Deploy RPC designate by running the
```
Comment (cyrichardson) [inline doc/installing/deploy-designate.rst]: Designate

File: doc/installing/deploy-designate.rst [hunk 19]
```diff
@@ -0,0 +1,29 @@
+.. _deploy-designate:
+
+Deploy designate
+================
+
+After you complete the steps described in
+:ref:`prepare-installation` and
+:ref:`configure-dns-server-pool`, you can deploy designate on your
```
Comment (cyrichardson) [inline doc/installing/deploy-designate.rst]: Designate in

File: doc/installing/deploy-designate.rst [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+``scripts/deploy.sh`` script that is stored in this repository.
+The deploy script performs the following actions:
+
+- Detects the version of OpenStack.
+- Creates containers for the Berkeley Internet Name Domain (BIND)
+  9 backend and for designate on all infrastructure nodes.
+- Deploys BIND 9 and configures it to work with designate.
+- Creates a pool configuration at ``/etc/openstack_deploy/designate_pools.yml``
+  to set up designate to work with the installed BIND 9 software.
+- Deploy designate by using the default configuration.
```
Comment (cyrichardson) [inline doc/installing/deploy-designate.rst]: Designate

File: doc/installing/deploy-designate.rst [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+- Configure the firewall on the infrastructure hosts to enable
+  communication from infrastructure nodes IP addresses to BIND 9.
+
+To deploy designate, run the following script:
```
Comment (cyrichardson) [inline doc/installing/deploy-designate.rst]: RPC Designate?

File: doc/installing/index.rst [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+Designate Installation Guide
+============================
+
+*Last updated:* |today|
+
+.. Define |product name| in conf.py
+.. You can define other variables in that file and use them throughout
+   your documents.
+
+``rpc-designate`` simplifies installation of multi-tenant
```
Comment (cyrichardson) [inline doc/installing/index.rst]: RPC Designate?

multitenant

File: doc/installing/prepare-installation.rst [hunk 23]
```diff
@@ -0,0 +1,122 @@
+.. _prepare-installation:
+
+Prepare for designate installation
+==================================
+
+Before deploying designate, you need to configure a few important components,
```
Comment (cyrichardson) [inline doc/installing/prepare-installation.rst]: RPC Designate?

File: doc/installing/prepare-installation.rst [hunk 24]
```diff
@@ -0,0 +1,122 @@
...
+including a DNS server, ports, and an Remote Name Daemon Control (RNDC)
+key. Then, create a DNS servers pool as described in
+:ref:`configure-dns-server-pool` and then, run the designate
```
Comment (cyrichardson) [inline doc/installing/prepare-installation.rst]: and  run the Designate

File: doc/installing/prepare-installation.rst [hunk 25]
```diff
@@ -0,0 +1,122 @@
...
+deployment script.
+
+.. _setup-dns-backend:
+
+Set up a DNS backend
+~~~~~~~~~~~~~~~~~~~~
+
+Although designate provides drivers to support different DNS
+backends, RPC supports BIND 9 only. You can either use an existing DNS
+deployment or set up new DNS nodes to work with designate.
```
Comment (cyrichardson) [inline doc/installing/prepare-installation.rst]: Designate?

File: doc/installing/prepare-installation.rst [hunk 26]
```diff
@@ -0,0 +1,122 @@
...
+To set up a DNS backend, follow these steps:
+
+#. Enable the creation of new availability zone by configuring the
+   following BIND 9 option:
+
+   .. code-block:: yaml
+
+      allow-new-zones yes;
+
+#. Enable designate to communicate with BIND 9 by creating
```
Comment (cyrichardson) [inline doc/installing/prepare-installation.rst]: Designate?

File: doc/installing/prepare-installation.rst [hunk 27]
```diff
@@ -0,0 +1,122 @@
...
+      };
+
+   Fore more information, see :ref:`designate-rndc`.
+
+.. _designate-ports:
+
+Configure network ports
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Designate must communicate bi-directionally with the
```
Comment (cyrichardson) [inline doc/installing/prepare-installation.rst]: bidirectionally

File: doc/installing/prepare-installation.rst [hunk 28]
```diff
@@ -0,0 +1,122 @@
...
+infrastructure nodes that run the designate containers and
```
Comment (cyrichardson) [inline doc/installing/prepare-installation.rst]: Designate?

File: doc/installing/prepare-installation.rst [hunk 29]
```diff
@@ -0,0 +1,122 @@
...
+
+       DNS -> OpenStack infrastructure nodes.
+
+   * - DNS servers
+     - * 53/UDP
+       * 53/TCP
+       * 953/TCP
+     -  DNS servers need the following ports to be open:
+
+        * 53/UDP -> Open for all traffic for DNS resolution
```
Comment (cyrichardson) [inline doc/installing/prepare-installation.rst]: (Periods at end of the bulleted list items to match the last one.)

File: doc/installing/prepare-installation.rst [hunk 30]
```diff
@@ -0,0 +1,122 @@
...
+
+        OpenStack Infra nodes -> DNS
+
+.. _designate-rndc:
+
+RNDC key
+~~~~~~~~
+
+RNDC enables configuration for DNS servers with the BIND 9 backend.
+In different operating system the RNDC key file is either
```
Comment (cyrichardson) [inline doc/installing/prepare-installation.rst]: systems, the

File: doc/installing/prepare-installation.rst [hunk 31]
```diff
@@ -0,0 +1,122 @@
...
+
+   * - Operating system
+     - Description
+   * - Ubuntu
+     - The BIND 9 package automatically generates an ``rndc.key``
+       file in the /etc/bind directory.
+   * - CentOS/RHEL
+     - You need to generate the key file by using the ``rndc-confgen``
+       command. This command creates the key file and places it in
+       the ``/etc`` directory
```
Comment (cyrichardson) [inline doc/installing/prepare-installation.rst]: directory.

File: doc/CONTRIBUTING.md (hunk 5 above)
Comment (svetkars) [reply doc/CONTRIBUTING.md]: rephrased for this repository.

File: doc/conf.py (hunk 9 above)
Comment (svetkars) [reply doc/conf.py]: leaving just the product name for now

File: doc/conf.py (hunk 10 above)
Comment (svetkars) [reply doc/conf.py]: I think we can delete one of the occurrences.

File: doc/index.rst (hunk 12 above)
Comment (svetkars) [reply doc/index.rst]: Will change to RPC Designate everywhere.

File: doc/installing/configure-dns-server-pool.rst (hunk 16 above)
Comment (svetkars) [reply doc/installing/configure-dns-server-pool.rst]: rephrased.

File: doc/installing/prepare-installation.rst (hunk 30 above)
Comment (svetkars) [reply doc/installing/prepare-installation.rst]: finished the sentence

**Pull request merged**